### PR TITLE
Updated with native speechT5 Model changes

### PR DESCRIPTION
**Pull request by AnishaUdayakumar**
Updated with native speechT5 Model changes including app.py and requirements.txt, removed bark_utils.py

### CODE REVIEW

File: recipes/conversational_voice_agent/app.py [hunk 1]
```diff
@@ -146,16 +153,44 @@ def chat(history: List) -> List[List[str]]:
...
     """
+    start_time = time.time()  # Start time
     prompt = conversation[-1][-1]
-    # generate audio for the last prompt (bot's response)
-    return SAMPLE_RATE, tts_model.generate_audio(prompt)
+
+    # Function to split text into sentences
+    def split_into_sentences(text):
+        sentences = re.split(r'(?<=[^A-Z].[.!?]) +(?=[A-Z])', text)
+        return [sentence.strip() for sentence in sentences if sentence.strip()]
```
Comment (adrianboguszewski) [inline recipes/conversational_voice_agent/app.py:172]: Use nltk package here

File: recipes/conversational_voice_agent/app.py [hunk 2]
```diff
@@ -146,16 +153,44 @@ def chat(history: List) -> List[List[str]]:
...
+
+    # Split the prompt into sentences
+    sentences = split_into_sentences(prompt)
+    audio_segments = []
+
+    for sentence in sentences:
+        inputs = tts_processor(text=sentence, return_tensors="pt")
+
+        # Check if the token count is within the limit
+        if inputs.input_ids.size(1) <= 600:
```
Comment (adrianboguszewski) [inline recipes/conversational_voice_agent/app.py:182]: What happens if the input is longer?

File: recipes/conversational_voice_agent/app.py [hunk 3]
```diff
@@ -146,16 +153,44 @@ def chat(history: List) -> List[List[str]]:
...
+            embeddings_dataset = load_dataset("Matthijs/cmu-arctic-xvectors", split="validation")
+            speaker_embeddings = torch.tensor(embeddings_dataset[7306]["xvector"]).unsqueeze(0)
```
Comment (adrianboguszewski) [inline recipes/conversational_voice_agent/app.py:184]: This is very inefficient. You should load it once, not in every loop.

File: recipes/conversational_voice_agent/app.py [hunk 4]
```diff
@@ -146,16 +153,44 @@ def chat(history: List) -> List[List[str]]:
...
+            speech = tts_model.generate_speech(inputs["input_ids"], speaker_embeddings, vocoder=tts_vocoder)
+            audio_segments.append(speech.numpy())
+
+    # Combine audio segments
+    combined_audio = np.concatenate(audio_segments, axis=0) if audio_segments else np.array([])
```
Comment (adrianboguszewski) [inline recipes/conversational_voice_agent/app.py:189]: This "if" seems to be unnecessary. `audio_segments` always exists.

File: recipes/conversational_voice_agent/load_tts.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import nltk
+from nltk.tokenize import word_tokenize
+
+# Attempt to download the 'punkt' tokenizer
+try:
+    nltk.data.find('tokenizers/punkt')
+except LookupError:
+    nltk.download('punkt')
```
Comment (adrianboguszewski) [inline recipes/conversational_voice_agent/load_tts.py:8]: Put it into separate function and call it from main

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]
Good job @AnishaUdayakumar. It works good. Please see the comments below

File: recipes/conversational_voice_agent/app.py (hunk 2 above)
Comment (AnishaUdayakumar) [reply recipes/conversational_voice_agent/app.py:206]: I will address by implementing the chunk_sentence function. This function breaks down sentences that exceed the token limit into smaller parts. Each chunk is then separately processed by the TTS model.

File: recipes/conversational_voice_agent/app.py (hunk 4 above)
Comment (AnishaUdayakumar) [reply recipes/conversational_voice_agent/app.py]: The revised synthesize function directly concatenates audio_segments without the conditional check.

File: recipes/conversational_voice_agent/load_tts.py (hunk 5 above)
Comment (AnishaUdayakumar) [reply recipes/conversational_voice_agent/load_tts.py]: Moved the NLTK to app.py

**Pull request closed without merging**